### PR TITLE
Issue #843/1946240: Remove the hardcoded 0 index in theme_status_message...

### PR DESCRIPTION
**Pull request by serundeputy**
This fixes: #843 / #1946240 Drupal 7.36 update meta issue.

### CODE REVIEW

File: core/modules/system/system.module [hunk 1]
```diff
@@ -3496,6 +3496,7 @@ function system_autoload_info() {
     'Archive_Tar' => 'system.tar.inc',
     'ModuleUpdater' => 'system.updater.inc',
     'ThemeUpdater' => 'system.updater.inc',
+    'BackdropSetMessageTest' => 'tests/system.test',
```
Comment (serundeputy) [inline core/modules/system/system.module]: @quicksketch I wasn't sure here; no other classes in the `tests/system.test` file seem to be mentioned in any `hook_autoload_info()` as far as I can tell.

So either this one shouldn't be either; or they all need to be added?

Comment (quicksketch) [inline core/modules/system/system.module]: Hi @serundeputy! You're right, test classes don't need to be added to `hook_autoload_info()`. The classes are loaded individually based on the class name and file path from the `*.tests.info` file.

File: core/modules/system/tests/system.test [hunk 2]
```diff
@@ -2360,6 +2360,35 @@ class SystemValidTokenTest extends BackdropWebTestCase {
...
+ */
+class BackdropSetMessageTest extends BackdropWebTestCase {
+
+  public static function getInfo() {
+    return array(
+      'name' => 'Messages',
+      'description' => 'Tests that messages can be displayed using backdrop_set_message().',
+      'group' => 'System',
+    );
+  }
```
Comment (quicksketch) [inline core/modules/system/tests/system.test]: In Backdrop, we don't have `getInfo()` methods in the test classes. Instead move this information into `system.tests.info` to register the test.

File: core/modules/system/system.tests.info [hunk 3]
```diff
@@ -195,3 +195,9 @@ name = Token scanning
 description = Scan token-like patterns in a dummy text to check token scanning.
 group = System
 file = tests/system.test
+
+[BackdropSetMessageTest extends BackdropWebTestCase]
```
Comment (quicksketch) [inline core/modules/system/system.tests.info]: This should just be `[BackdropSetMessageTest]`

File: core/modules/system/system.tests.info [hunk 4]
```diff
@@ -195,3 +195,9 @@ name = Token scanning
...
+name = Messages
+description = Tests that messages can be displayed using backdrop_set_message().
+group = System
+file = tests/system.test
```
Comment (quicksketch) [inline core/modules/system/system.tests.info]: Add a new line at the bottom of the file.

File: core/modules/system/tests/system.test [hunk 5]
```diff
@@ -2360,6 +2360,27 @@ class SystemValidTokenTest extends BackdropWebTestCase {
...
+    parent::setUp('system_test');
+  }
+
+  /**
+   * Tests setting messages and removing one before it is displayed.
+   */
+  function testSetRemoveMessages() {
+    // The page at system-test/drupal-set-message sets two messages and then
+    // removes the first before it is displayed.
+    $this->drupalGet('system-test/drupal-set-message');
```
Comment (quicksketch) [inline core/modules/system/tests/system.test]: `drupalGet` => `backdropGet`

`drupal-set-message` => `backdrop-set-message`

**Comment by serundeputy** [comment]
@quicksketch updated; rebased;  also doh;

**Comment by quicksketch** [comment]
Super, thanks @serundeputy.

> System: Messages (BackdropSetMessageTest) 5 passes, 0 fails, 0 exceptions, and 1 debug message

Looks like the new tests got picked up and passed. Nice one. There's one more "drupal" reference left, and we could probably speed up these tests by using the testing profile.

File: core/modules/system/tests/system.test [hunk 6]
```diff
@@ -2360,6 +2360,27 @@ class SystemValidTokenTest extends BackdropWebTestCase {
 }
 
 /**
+ * Tests backdrop_set_message() and related functions.
+ */
+class BackdropSetMessageTest extends BackdropWebTestCase {
+
```
Comment (quicksketch) [inline core/modules/system/tests/system.test]: Let's add the testing profile here:

```
protected $profile = 'testing';
```

That'll make for a lighter testing installation for this trivial test.

File: core/modules/system/tests/system.test [hunk 7]
```diff
@@ -2360,6 +2360,27 @@ class SystemValidTokenTest extends BackdropWebTestCase {
...
+  function setUp() {
+    parent::setUp('system_test');
+  }
+
+  /**
+   * Tests setting messages and removing one before it is displayed.
+   */
+  function testSetRemoveMessages() {
+    // The page at system-test/drupal-set-message sets two messages and then
```
Comment (quicksketch) [inline core/modules/system/tests/system.test]: One last "drupal" => "backdrop".

File: core/modules/system/tests/system.test (hunk 6 above)
Comment (serundeputy) [inline core/modules/system/tests/system.test]: of course;  will do;

File: core/modules/system/tests/system.test (hunk 7 above)
Comment (serundeputy) [inline core/modules/system/tests/system.test]: what is wrong with me (rhetorical)?  of course i'll update.

**Comment by quicksketch** [comment]
Thanks! Merged in and cherry-picked back into 1.0.x.

**Pull request merged**